include/exadg/incompressible_navier_stokes/spatial_discretization/turbulence_model.cpp

```diff
@@ -36,27 +41,45 @@ TurbulenceModel<dim, Number>::initialize(
   dealii::MatrixFree<dim, Number> const &                matrix_free_in,
   dealii::Mapping<dim> const &                           mapping_in,
   std::shared_ptr<Operators::ViscousKernel<dim, Number>> viscous_kernel_in,
-  TurbulenceModelData const &                            data_in)
+  TurbulenceModelData const &                            turbulence_model_data_in,
+  unsigned int                                           dof_index_velocity_in,
+  unsigned int                                           quad_index_velocity_linear_in,
+  unsigned int                                           degree_velocity_in)
 {
-  matrix_free     = &matrix_free_in;
-  viscous_kernel  = viscous_kernel_in;
-  turb_model_data = data_in;
+  Base::initialize(matrix_free_in,
+                   viscous_kernel_in,
+                   dof_index_velocity_in,
+                   quad_index_velocity_linear_in);
+
+  turbulence_model_data = turbulence_model_data_in;
+  degree_velocity       = degree_velocity_in;
+
+  turbulence_model_data.check();
 
   calculate_filter_width(mapping_in);
 }
 
 template<int dim, typename Number>
 void
-TurbulenceModel<dim, Number>::calculate_turbulent_viscosity(VectorType const & velocity) const
+TurbulenceModel<dim, Number>::set_viscosity(VectorType const & velocity) const
+{
+  Base::viscous_kernel->set_constant_coefficient(Base::viscosity_newtonian_limit);
```

Comment:
related to comment https://github.com/exadg/exadg/pull/381/files#r1172249210: 
 Note that this function does not call `SpatialOperatorBase::set_constant_viscosity()`